src/app/gallery/page.tsx

```diff
@@ -22,35 +24,62 @@ import { useCmsClient } from "../client/restClient/cmsClient";
 
 export default function Gallery() {
   const [query, setQuery] = useState('');
+  const [data, setData] = useState([]);
+  const [categories, setCategories] = useState([]);
 
   const cmsClient = useCmsClient();
 
```

Comment:
First of all you never expose endpoints like this. we are storing them in Env variables. So that they can be accessable only in build time. 
now all our enpoints are public so it is not big issue, but keep in mind that is bad practice!  You should never expose this. 
Aslo CmsClient is  instatiated like this 
``` const cmsClient = useCmsClient();```
This is also bad approach because then you will have to instantiate client mulitple times. 

So to not be long what is going here you can read provider.